pkg/networkservice/utils/metadata/client.go

```diff
@@ -37,12 +37,16 @@ func NewClient() networkservice.NetworkServiceClient {
 }
 
 func (m *metaDataClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
-	ctx = store(ctx, request.GetConnection().GetId(), &m.Map)
+	connID := request.GetConnection().GetId()
```

Comment:
Good catch!

Reply:
Although... it should *always* be true that the conn..GetId() below is the same as the request.GetConnection().GetId() ... its *good* defensive coding what you've done here, but an issue here would have been a sign of a bug in updatetoken ...